## Test de hipótesis en un regresión lineal

Analizaremos el set de datos “Boston Houssing“ en R (del paquete mlbench)

El set de datos cuenta con 14 variables, de las cuales estudiaremos las siguientes:

- $X_1$: (nox) Concentración de óxido de nitrógeno (partes por 10 millones)
- $X_2$: (rm) Promedio de cantidad de habitaciones por vivienda
- $X_3$: (lstat) Status mínimo de la población (porcentaje)
- $Y$: (medv) Valor mediano de las casas ocupadas por dueños en miles de dólares

Se ajustó el modelo:

- $Y$ = $β_0$ + $β_1$ * $X_1$ + $β_2$ * $X_2$ + $β_3$ * $X_3$ + $\epsilon$

**a)** ¿Qué puede decir a partir de la **tabla de correlaciones entre las variables**? Si tuviera que elegir
una sola variable para plantear el modelo, ¿Cuál eligiría?

**b)** A partir de la tabla de coeficientes estimados, ¿Qué **variables** son **significativas**? ¿A qué nivel?
¿Cuál es el valor de s? Especificar las **hipótesis nulas y alternativas** de cada uno de los test t reportados en la tabla. ¿Cómo se calculan el valor p de este test?

**c)** ¿Es la **regresión significativa**? Especificar las **hipótesis nula y alternativa de este test**. ¿Cómo se
calcula el **p-valor** en este caso? ¿Rechazaría a un **nivel de significación** de 0.05?

**d)** ¿Cómo se calcula la **matriz de correlación de los estimadores**?

**e)** ¿Qué **supuestos** necesita realizar para que las conclusiones anteriores sean válidas?

In [2]:
library(mlbench)
data(BostonHousing)

In [4]:
BostonHousing

crim     zn   indus chas nox   rm    age   dis    rad tax ptratio b     
1   0.00632  18.0 2.31  0    0.538 6.575  65.2 4.0900 1   296 15.3    396.90
2   0.02731   0.0 7.07  0    0.469 6.421  78.9 4.9671 2   242 17.8    396.90
3   0.02729   0.0 7.07  0    0.469 7.185  61.1 4.9671 2   242 17.8    392.83
4   0.03237   0.0 2.18  0    0.458 6.998  45.8 6.0622 3   222 18.7    394.63
5   0.06905   0.0 2.18  0    0.458 7.147  54.2 6.0622 3   222 18.7    396.90
6   0.02985   0.0 2.18  0    0.458 6.430  58.7 6.0622 3   222 18.7    394.12
7   0.08829  12.5 7.87  0    0.524 6.012  66.6 5.5605 5   311 15.2    395.60
8   0.14455  12.5 7.87  0    0.524 6.172  96.1 5.9505 5   311 15.2    396.90
9   0.21124  12.5 7.87  0    0.524 5.631 100.0 6.0821 5   311 15.2    386.63
10  0.17004  12.5 7.87  0    0.524 6.004  85.9 6.5921 5   311 15.2    386.71
11  0.22489  12.5 7.87  0    0.524 6.377  94.3 6.3467 5   311 15.2    392.52
12  0.11747  12.5 7.87  0    0.524 6.009  82.9 6.2267 5   311 15.2    396.90
13  0.09378  12.5 7.87  0    0.524 5.889  39.0 5.4509 5   311 15.2    390.50
14  0.62976   0.0 8.14  0    0.538 5.949  61.8 4.7075 4   307 21.0    396.90
15  0.63796   0.0 8.14  0    0.538 6.096  84.5 4.4619 4   307 21.0    380.02
16  0.62739   0.0 8.14  0    0.538 5.834  56.5 4.4986 4   307 21.0    395.62
17  1.05393   0.0 8.14  0    0.538 5.935  29.3 4.4986 4   307 21.0    386.85
18  0.78420   0.0 8.14  0    0.538 5.990  81.7 4.2579 4   307 21.0    386.75
19  0.80271   0.0 8.14  0    0.538 5.456  36.6 3.7965 4   307 21.0    288.99
20  0.72580   0.0 8.14  0    0.538 5.727  69.5 3.7965 4   307 21.0    390.95
21  1.25179   0.0 8.14  0    0.538 5.570  98.1 3.7979 4   307 21.0    376.57
22  0.85204   0.0 8.14  0    0.538 5.965  89.2 4.0123 4   307 21.0    392.53
23  1.23247   0.0 8.14  0    0.538 6.142  91.7 3.9769 4   307 21.0    396.90
24  0.98843   0.0 8.14  0    0.538 5.813 100.0 4.0952 4   307 21.0    394.54
25  0.75026   0.0 8.14  0    0.538 5.924  94.1 4.3996 4   307 21.0    394.33
26  0.84054   0.0 8.14  0    0.538 5.599  85.7 4.4546 4   307 21.0    303.42
27  0.67191   0.0 8.14  0    0.538 5.813  90.3 4.6820 4   307 21.0    376.88
28  0.95577   0.0 8.14  0    0.538 6.047  88.8 4.4534 4   307 21.0    306.38
29  0.77299   0.0 8.14  0    0.538 6.495  94.4 4.4547 4   307 21.0    387.94
30  1.00245   0.0 8.14  0    0.538 6.674  87.3 4.2390 4   307 21.0    380.23
⋮   ⋮        ⋮    ⋮     ⋮    ⋮     ⋮     ⋮     ⋮      ⋮   ⋮   ⋮       ⋮     
477  4.87141 0    18.10 0    0.614 6.484 93.6  2.3053 24  666 20.2    396.21
478 15.02340 0    18.10 0    0.614 5.304 97.3  2.1007 24  666 20.2    349.48
479 10.23300 0    18.10 0    0.614 6.185 96.7  2.1705 24  666 20.2    379.70
480 14.33370 0    18.10 0    0.614 6.229 88.0  1.9512 24  666 20.2    383.32
481  5.82401 0    18.10 0    0.532 6.242 64.7  3.4242 24  666 20.2    396.90
482  5.70818 0    18.10 0    0.532 6.750 74.9  3.3317 24  666 20.2    393.07
483  5.73116 0    18.10 0    0.532 7.061 77.0  3.4106 24  666 20.2    395.28
484  2.81838 0    18.10 0    0.532 5.762 40.3  4.0983 24  666 20.2    392.92
485  2.37857 0    18.10 0    0.583 5.871 41.9  3.7240 24  666 20.2    370.73
486  3.67367 0    18.10 0    0.583 6.312 51.9  3.9917 24  666 20.2    388.62
487  5.69175 0    18.10 0    0.583 6.114 79.8  3.5459 24  666 20.2    392.68
488  4.83567 0    18.10 0    0.583 5.905 53.2  3.1523 24  666 20.2    388.22
489  0.15086 0    27.74 0    0.609 5.454 92.7  1.8209  4  711 20.1    395.09
490  0.18337 0    27.74 0    0.609 5.414 98.3  1.7554  4  711 20.1    344.05
491  0.20746 0    27.74 0    0.609 5.093 98.0  1.8226  4  711 20.1    318.43
492  0.10574 0    27.74 0    0.609 5.983 98.8  1.8681  4  711 20.1    390.11
493  0.11132 0    27.74 0    0.609 5.983 83.5  2.1099  4  711 20.1    396.90
494  0.17331 0     9.69 0    0.585 5.707 54.0  2.3817  6  391 19.2    396.90
495  0.27957 0     9.69 0    0.585 5.926 42.6  2.3817  6  391 19.2    396.90
496  0.17899 0     9.69 0    0.585 5.670 28.8  2.7986  6  391 19.2    393.29


In [3]:
nox <- BostonHousing$nox 

In [4]:
rm <- BostonHousing$rm

In [5]:
lstat <- BostonHousing$lstat

In [14]:
medv <- BostonHousing$medv

**a) ¿Qué puede decir a partir de la tabla de correlaciones entre las variables? Si tuviera que elegir
una sola variable para plantear el modelo, ¿Cual eligiría?**

In [7]:
cor(cbind(medv,nox,rm,lstat))

medv       nox        rm         lstat     
medv   1.0000000 -0.4273208  0.6953599 -0.7376627
nox   -0.4273208  1.0000000 -0.3021882  0.5908789
rm     0.6953599 -0.3021882  1.0000000 -0.6138083
lstat -0.7376627  0.5908789 -0.6138083  1.0000000

Si tuviera que elegir una para predecir **medv** (el valor mediano de las casas) elijo la variable que tiene **mayor correlación en módulo** con medv por lo que escogería la variable predictora **lstat** (status mínimo de la población).

**Una observación:** 

Si las $X$ (variables predictoras) están muy correlacionadas entre si, entonces aportan la misma información. En particular si una es combinación lineal de las otras la matriz de diseño X no será rango completo.  Dos variables predictoras están muy correlacionadas si hay aproximadamente un 0.95 de correlación entre ellas.

**b) A partir de la tabla de coeficientes estimados, ¿Qué variables son significativas? ¿A qué nivel?
¿Cuál es el valor de s? Especificar las hipótesis nulas y alternativas de cada uno de los test t reportados en la tabla. ¿Cómo se calculan el valor p de este test?**

In [10]:
ajuste <- lm(medv ~ lstat + nox + rm)

In [12]:
summary(ajuste)


Call:
lm(formula = medv ~ lstat + nox + rm)

Residuals:
     Min       1Q   Median       3Q      Max 
-17.7048  -3.4950  -0.9992   1.9397  28.4939 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.76669    3.28617  -0.233    0.816    
lstat       -0.62288    0.05193 -11.994   <2e-16 ***
nox         -1.84563    2.65052  -0.696    0.487    
rm           5.12433    0.44671  11.471   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.543 on 502 degrees of freedom
Multiple R-squared:  0.6389,	Adjusted R-squared:  0.6368 
F-statistic: 296.1 on 3 and 502 DF,  p-value: < 2.2e-16


Consideraremos que una variable predictora $X_i$ **es significativa si no podemos rechazar la hipótesis nula** del siguiente test: 

$H_0$: $\beta_i$ = 0 $\hspace{10mm}$ vs $\hspace{10mm}$  $H_1$: $\beta_i$ $\neq$ 0

Observamos que $\beta_i$ es el coeficiente que multiplica a $X_i$ en el modelo de regresión lineal.

El test de hipótesis que se plantea para cada $\beta_i$ es el siguiente:

$\delta(X)$ =$\mathbb{1}${$ |T|> t_{n-p,\frac{\alpha}{2}} $}

Donde: 

- $T$ = $\frac{\hat{\beta_i}} {S\sqrt{d_{i,i}}}$

**Observación** el estadistico que usamos es:
$\frac{\hat{\beta_i}} {S\sqrt{d_{i,i}}}$ 
De manera que bajo $H_0$ verdadera (es decir $\beta_i= 0$),  $T$ distribuye t de student de n-p grados de libertad

- $d_{i,i}$ : elemento de la diagonal de ${(X^T X)}^{-1}$

- $\hat{\beta_i}$ = estimación del parametro $\beta_i$


De el resumen de la regresión realizada observamos: 

Llamando $\beta_1$ al parametro de la variable 'nox'

- $\hat{\beta_1}$ = -1.84 $\hspace{3mm}$ Es el coeficiente estimado para $\beta$

- $S\sqrt{d_{1,1}}$ = 2.65  $\hspace{3mm}$ es el 'Standard Error'

- t value = $\hat{\beta_1}$ / $(S\sqrt{d_{1,1}})$ = -0.69 $\hspace{3mm}$ (Es el valor del estadístico T para esta estimacion)
    
    
Ahora nos fijamos si este t value es mayor que el cuantil $t_{n-p, 1-\frac{\alpha}{2}}$ (n-p = 506 - 4) 

Tomando alfa = 0.05:


In [13]:
n <- 506
p <- 4
alpha <- 0.05
qt(1-(alpha)/2, n-p)

[1] 1.964701

Dado que mi **t-value es menor que el cuantil** digo que **no tengo suficiente información para rechazar la hipótesis nula** y por lo tanto considero que mi varible **nox es significativa**. **Observación:** esto no quiere decir que $\beta_1$ no sea 0. La conclusión fuerte de un test de hipótesis siempre resulta de rechazar la hipótesis nula. Es decir, si la rechazo puedo asegurar que es 0, pero si no la rechazo no puedo asegurar que no sea 0.

**Notas para mi:** Realizar el mismo T test para las otras variables. Automatizarlo.

parte c)
La regresion es significativa si al menos algun coeficiente (**excepto el intercept** <- importante para armar la matriz C del F-TEST) es distinto de 0 (este es el F test)